src/Errors.huff

```diff
@@ -0,0 +1,112 @@
+#define constant NO_SELFDESTRUCT_LENGTH = 0x0F
+
+// "UNUSED"
+#define constant UNUSED_ERROR = 0x554e555345440000000000000000000000000000000000000000000000000000
+#define constant UNUSED_LENGTH = 0x06
+
+// "NOT_OWNER"
+#define constant NOT_OWNER_ERROR = 0x4e4f545f4f574e45520000000000000000000000000000000000000000000000
+#define constant NOT_OWNER_LENGTH = 0x09
+
+// "CHECK_UNAUTHORIZED"
+#define constant CHECK_UNAUTHORIZED_ERROR = 0x434845434b5f554e415554484f52495a45440000000000000000000000000000
+#define constant CHECK_UNAUTHORIZED_LENGTH = 0x0C
+
+// "VERIFICATION_FAILED"
+#define constant VERIFICATION_FAILED_ERROR = 0x564552494649434154494f4e5f4641494c454400000000000000000000000000
+#define constant VERIFICATION_FAILED_LENGTH = 0x0D
+
+/* -------------------------------------------------------------------------- */
+/*                                   ERRORS                                   */
+/* -------------------------------------------------------------------------- */
+
+/// @notice Reverts with an "JUMP_FAIL" message
+#define macro JUMP_FAIL() = takes (0) returns (0) {
+    [JUMP_FAIL_ERROR]
+    [JUMP_FAIL_LENGTH]
+    REQUIRE()
+}
+
+/// @notice Reverts with an "NO_SELFDESTRUCT" message
+#define macro NO_SELFDESTRUCT() = takes (0) returns (0) {
+    [NO_SELFDESTRUCT_ERROR]
+    [NO_SELFDESTRUCT_LENGTH]
+    REQUIRE()
+}
+
+/// @notice Reverts with an "UNUSED" message
+#define macro UNUSED() = takes (0) returns (0) {
+    [UNUSED_ERROR]
+    [UNUSED_LENGTH]
+    REQUIRE()
+}
+
+/// @notice Reverts with an "NOT_OWNER" message
+#define macro NOT_OWNER() = takes (0) returns (0) {
+    [NOT_OWNER_ERROR]
+    [NOT_OWNER_LENGTH]
+    REQUIRE()
+}
+
+/// @notice Reverts with an "CHECK_UNAUTHORIZED" message
+#define macro CHECK_UNAUTHORIZED() = takes (0) returns (0) {
+    [CHECK_UNAUTHORIZED_ERROR]
+    [CHECK_UNAUTHORIZED_LENGTH]
+    REQUIRE()
+}
+
+/// @notice Reverts with an "VERIFICATION_FAILED" message
+#define macro VERIFICATION_FAILED() = takes (0) returns (0) {
+    [VERIFICATION_FAILED_ERROR]
+    [VERIFICATION_FAILED_LENGTH]
+    REQUIRE()
+}
+
+/* -------------------------------------------------------------------------- */
+/*                                   MACROS                                   */
+/* -------------------------------------------------------------------------- */
+
+/// @dev Solidity require(...) 
+/// 
+/// MEMORY LAYOUT WHEN THROWN
+/// | sig    || message offset                                                 || message length                                                 || message "revert"                                               |
+/// 0x08c379a  0000000000000000000000000000000000000000000000000000000000000020  0000000000000000000000000000000000000000000000000000000000000006  7265766572740000000000000000000000000000000000000000000000000000
+///
+/// Error `string` MUST be no greater than 32 bytes.
+#define macro REQUIRE() = takes (3) returns (0) {
```

Comment:
Maybe we could add a test for this one ?
It seems pretty important and from what I saw with errors, it seems there is a mismatch in input provided and what it takes.